shadowsocks-csharp/Model/Configuration.cs

```diff
@@ -78,6 +82,10 @@ public static Configuration Load()
                 string configContent = File.ReadAllText(CONFIG_FILE);
                 Configuration config = JsonConvert.DeserializeObject<Configuration>(configContent);
                 config.isDefault = false;
+                if (UpdateChecker.Asset.CompareVersion(UpdateChecker.Version, config.version) > 0)
+                {
+                    config.updated = true; 

```

Comment:
How about launch an event such as "OnUpdated" rather than a flag?
So we can register events such as: 1. show balloon. 2. run migration script when necessary 3. show change log if necessary

Reply:
Good.